### PR TITLE
[CI:DOCS] Clarify that MacOSX docs are client specific

### PR DESCRIPTION
**Pull request by jesselang**
Seems helpful to mention here that only the client can be built directly on MacOSX.

Signed-off-by: Jesse Lang <jesse@jesselang.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ the “In Progress” label be set and a member will do so for you.
 
 ## Contributing to Podman
 
-This section describes how to start a contribution to Podman. These instructions are geared towards using a Linux development machine, or doing development on the podman backend. For instructions on working on MacOS, read the [docs for building on osx](./build_osx.md).
+This section describes how to start a contribution to Podman. These instructions are geared towards using a Linux development machine, which is required for doing development on the podman backend. Check out these instructions for [building the podman client on MacOSX](./build_osx.md).
```
Comment (mheon) [inline CONTRIBUTING.md:45]: Linter is complaining about trailing whitespace on this line

**Comment by rhatdan** [comment]
Tests are complaining about excess whitespace.

**Comment by rhatdan** [comment]
Can you squash your commits, to get passed whitespace problem

git rebase -i origin
git push --force

**Comment by jesselang** [comment]
Squashed as requested. Tests seem to be passing now.

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
Thanks @jesselang

**Pull request merged**